src/lib/network_pool/indexed_pool.ml

```diff
@@ -51,8 +52,14 @@ let currency_consumed :
     fee_amt
     +
     match cmd'.payload.body with
-    | Payment {amount; _} ->
-        amount
+    | Payment {amount; receiver} ->
+        if
+          Token_id.equal cmd'.payload.common.fee_token
+            (Account_id.token_id receiver)
+        then
+          (* The fee-payer is also the sender account, include the amount. *)
+          amount
+        else zero
```

Comment:
I've added a comment to clarify here. The amount is only relevant to the fee-payer's balance when they are the one sending the transaction; otherwise, the rest of the balance changes affect the source's balance.

For the account creation fee, we cannot know at indexing time which transaction will end up creating the account, so we end up violating the invariants that the indexed pool is supposed to guarantee. My preferred solution is to charge the base fee but not move any funds if paying the creation fee would fail -- similar to what we do when `source =/= fee_payer` and the source doesn't have enough when the transaction is executed. I've opened an issue https://github.com/CodaProtocol/coda/issues/4500.